addons/test_mail/tests/test_mail_activity.py

```diff
@@ -141,6 +141,23 @@ def _employee_crash(*args, **kwargs):
                     'test_mail.mail_act_test_todo',
                     user_id=self.user_admin.id)
 
+        test_activity.user_id = self.user_employee
+        test_activity.flush_recordset()
+
+        # user can read activities assigned to him even if he has no access to the document
+        with patch.object(MailTestActivity, 'check_access_rights', autospec=True, side_effect=_employee_crash):
+            searched_activity = self.env['mail.activity'].with_user(self.user_employee).search(
+                [('id', '=', test_activity.id)])
+            searched_activity.read(['summary'])
```

Comment:
Would add a check we effectively have the right result
```suggestion
            self.assertEqual(found, test_activity)
            searched_activity.read(['summary'])
```

Reply:
Yes indeed, thanks